lib/Dialect/ArithExt/IR/ArithExtOps.td

```diff
@@ -0,0 +1,48 @@
+#ifndef LIB_DIALECT_ARITHEXT_IR_ARITHEXTOPS_TD_
+#define LIB_DIALECT_ARITHEXT_IR_ARITHEXTOPS_TD_
+
+include "lib/Dialect/ArithExt/IR/ArithExtDialect.td"
+include "mlir/IR/BuiltinAttributes.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+
+class ArithExt_Op<string mnemonic, list<Trait> traits = [Pure]> :
+        Op<ArithExt_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::arith_ext";
+}
+
+def ArithExt_BarrettReduceOp : ArithExt_Op<"barrett_reduce", [SameOperandsAndResultType]> {
+  let summary = "Compute the first step of the Barrett reduction.";
```

Comment:
I think `arith_ext` ops should be `ElementwiseMappaple` wherever possible, and I don't think it'd be very complicated to enable this here, as the lowerings are to `affine` which is already `ElementwiseMappaple` + the ops already take `SinglessIntegerLike`, i.e., also permit tensors/etc.